### PR TITLE
Add gift card creation

### PR DESCRIPTION
**Pull request by mmarkusik**
I want to merge this change because it adds creating single gift card

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3910 Add single Gift Card creation](https://app.clickup.com/t/2549495/SALEOR-3910)

File: src/components/ContentOrSkeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+interface ContentOrSkeletonProps {
+  condition: any;
+  children: React.ReactNode;
+}
+
+const ContentOrSkeleton: React.FC<ContentOrSkeletonProps> = ({
+  condition,
+  children
+}) => (!!condition ? <>{children}</> : <Skeleton />);
```
Comment (orzechdev) [inline src/components/ContentOrSkeleton.tsx:12]: I don't like this component. It's good to hide conditional displaying of skeleton, but I would rather think that clearly the main purpose of this component should be to display content, while showing skeleton should be temporary state. Currently the name `ContentOrSkeleton` doesn't suggest it, I would rather see e.g. `WithSkeleton`. Also `condition` prop is not precise, as we don't know clearly what occurs when it is true or false, without seeing implementation details. An interface should clearly define what it expects, e.g. `showSkeleton`.

But going even further, what do you think about even refactoring current `Skeleton` component in `@saleor/components/Skeleton` to be able to accept `show` prop and children, and then use it like:
```tsx
<Skeleton show={!loading}>
  <SomeChildren />
</Skeleton>
```
or use it like it is now when `children` are absent and `show` defaults to true:
```tsx
<Skeleton />
```
?

Comment (orzechdev) [reply src/components/ContentOrSkeleton.tsx:12]: Hiding conditional displaying of the skeleton is a very good idea 👍 .

**Review by orzechdev — CHANGES_REQUESTED** [review]

File: src/components/ContentOrSkeleton.tsx (hunk 1 above)
Comment (mmarkusik) [reply src/components/ContentOrSkeleton.tsx:12]: Sounds good to me, @jwm0 does adding this ability to exsiting skeleton sounds ok to you?

File: src/giftCards/GiftCardsList/GiftCardsListTable/GiftCardsListTable.tsx [hunk 2]
```diff
@@ -7,50 +7,48 @@ import {
...
-    // TEMP
-    // disabled
-  };
+  const {
+    toggle,
+    isSelected,
+    giftCards,
+    numberOfColumns,
+    loading
+  } = useContext(GiftCardsListContext);
```
Comment (jwm0) [inline src/giftCards/GiftCardsList/GiftCardsListTable/GiftCardsListTable.tsx:42]: Didn't we want to export a hook: `useGiftCardBulkActions` that hooks into this context?

File: src/giftCards/GiftCardsList/GiftCardsListTable/GiftCardsListTableHeader.tsx [hunk 3]
```diff
@@ -9,25 +9,25 @@ import Label, {
...
-  const { toggleAll, listElements, giftCards } = useContext(
-    GiftCardsListContext
-  );
+  const {
+    toggleAll,
+    listElements,
+    giftCards,
+    numberOfColumns,
+    loading
+  } = useContext(GiftCardsListContext);
```
Comment (jwm0) [inline src/giftCards/GiftCardsList/GiftCardsListTable/GiftCardsListTableHeader.tsx:30]: same as above

File: src/giftCards/components/GiftCardExpirySelect/GiftCardExpirySelect.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  {
+    label: messages.expirationDateLabel,
+    value: GiftCardExpiryTypeEnum.EXPIRY_DATE
+  }
+];
+
+interface GiftCardExpirySelectProps {
+  change: FormChange;
+  expiryPeriodType: TimePeriodTypeEnum;
+  expiryPeriodAmount: string;
```
Comment (jwm0) [inline src/giftCards/components/GiftCardExpirySelect/GiftCardExpirySelect.tsx:41]: Should it be a string? Perhaps we should cast it later to a string but it makes more sense to have it as a number to me?

File: src/giftCards/GiftCardUpdatePage/messages.ts [hunk 5]
```diff
@@ -1,12 +1,34 @@
...
   }
 });
+
+export function getGiftCardErrorMessage(
+  error: Omit<GiftCardUpdate_giftCardUpdate_errors, "__typename"> | undefined,
+  intl: IntlShape
+): string {
+  if (error) {
+    switch (error.code) {
+      case GiftCardErrorCode.GRAPHQL_ERROR:
```
Comment (jwm0) [inline src/giftCards/GiftCardUpdatePage/messages.ts:21]: This doesn't seem like a good error for input level field 🤔

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/components/TextWithSelectField/TextWithSelectField.tsx [hunk 6]
```diff
@@ -3,47 +3,100 @@ import SingleSelectField, {
...
+  const {
+    name: selectFieldName,
+    value: selectFieldValue,
+    className: selectFieldClassName
+  } = selectFieldProps;
+
+  // in case one of the fields in the form is empty
+  // we need to save the other part of the field as well
+  const handleChange = (type: "selectField" | "textField") => (
+    event: React.ChangeEvent<any>
```
[the remark this answers is not in the document]
Comment (mmarkusik) [reply src/components/TextWithSelectField/TextWithSelectField.tsx:56]: Actually I think our own `ChangeEvent` is even better

File: src/giftCards/GiftCardUpdatePage/messages.ts (hunk 5 above)
Comment (mmarkusik) [reply src/giftCards/GiftCardUpdatePage/messages.ts:21]: We use this logic in multiple places, maybe we should remove it from all of them then? 🤔

File: src/giftCards/components/GiftCardExpirySelect/GiftCardExpirySelect.tsx (hunk 4 above)
Comment (jwm0) [reply src/giftCards/components/GiftCardExpirySelect/GiftCardExpirySelect.tsx:41]: Is it also a String in schema? If not we should treat it as number everywhere and only cast it to String in the input itself

File: src/giftCards/GiftCardUpdatePage/messages.ts (hunk 5 above)
Comment (jwm0) [reply src/giftCards/GiftCardUpdatePage/messages.ts:21]: Create a ticket for this in 3.1/maintenance list and leave it as is right now

File: src/giftCards/components/GiftCardExpirySelect/GiftCardExpirySelect.tsx (hunk 4 above)
Comment (mmarkusik) [reply src/giftCards/components/GiftCardExpirySelect/GiftCardExpirySelect.tsx:41]: Not, then adding this as a number and parsing to the text with select field 👍

**Review by jwm0 — APPROVED** [review]

**Pull request merged**